src/main/java/org/apache/commons/lang3/LocaleUtils.java

```diff
@@ -295,14 +297,13 @@ public static boolean isAvailableLocale(final Locale locale) {
         }
         List<Locale> langs = cLanguagesByCountry.get(countryCode);
         if (langs == null) {
-            langs = new ArrayList<>();
-            final List<Locale> locales = availableLocaleList();
-            for (final Locale locale : locales) {
-                if (countryCode.equals(locale.getCountry()) &&
-                    locale.getVariant().isEmpty()) {
-                    langs.add(locale);
-                }
-            }
+            // Allows Locales to be processed using multiple threads.
+            langs = availableLocaleList().parallelStream()
+                    .filter(locale -> countryCode.equals(locale.getCountry()))
+                    .filter(locale -> locale.getVariant().isEmpty())
+                    .collect(Collectors.toList());
+            // Collectors.toUnmodifiableList() is an alternative, but it is only
+            // available starting with Java 10.
             langs = Collections.unmodifiableList(langs);
             cLanguagesByCountry.putIfAbsent(countryCode, langs);
```

Comment:
curious, could all of this be replaced with any of the `compute..` variants in `Map`?

Reply:
It should be possible, but I'm not sure how complex the resulting lambda will be.

Reply:
It's also not "simpler", it's just different, without even thinking about performance consequences....

Reply:
It does make use of parallelStream(), though.